CIP-0108/cip-0108.common.schema.json

```diff
@@ -0,0 +1,122 @@
+{
+    "title": "CIP-108 Common",
+    "description": "Metadata document for Cardano governance actions, extending CIP-100",
+    "type": "object",
+    "required": ["hashAlgorithm", "authors", "body"],
+    "properties": {
+        "hashAlgorithm": {
+            "$ref": "#/definitions/hashAlgorithm"
+        },
+        "authors": {
+            "$ref": "#/definitions/authors"
+        },
+        "body": {
+            "$ref": "#/definitions/body"
+        }
+    },
+    "definitions": {
+        "hashAlgorithm": {
+            "type": "string",
+            "enum": ["blake2b-256"],
+            "title": "Hash Algorithm",
+            "description": "The algorithm used to authenticate this document externally (CIP-100)"
+        },
+        "authors": {
+            "title": "Authors",
+            "description": "The authors of this governance metadata (CIP-100)",
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/author"
+            }
+        },
+        "author": {
+            "title": "Author",
+            "description": "An author endorsing the content of a metadata document (CIP-100)",
+            "type": "object",
+            "required": ["name", "witness"],
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "title": "Name"
+                },
+                "witness": {
+                    "$ref": "#/definitions/witness"
```

Comment:
You do not define witness in this document? I believe that instead this needs to be a link to CIP-100 witness definition? By the same token, the entire author/authors blocks could probably be shunted off to the CIP-100 definition unless there is something being customized here in CIP-108?

Reply:
good catch
I wasnt sure if I should add the CIP-100 related definitions to the common schema file or not

Reply:
Generally speaking they are fine to include as long as you reference them using a ref link that points to the canonical source. This just tells the schema checker to look at that link for that piece of data